app/frontend/src/features/search/useSearchFilters.ts

```diff
@@ -0,0 +1,88 @@
+import { Location } from "history";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+import { UserSearchFilters as ServiceUserSearchFilters } from "service/search";
+
+export interface SearchFilters extends ServiceUserSearchFilters {
+  location?: string;
+}
+
+export default function useSearchFilters(route: string) {
+  const location = useLocation();
+  const [active, setActive] = useState<SearchFilters>(
+    locationToFilters(location)
+  );
+  const pending = useRef<SearchFilters>(locationToFilters(location));
```

Comment:
```suggestion
  const pending = useRef<SearchFilters>(active);
```
Since I believe `active` will only be used for initialising the ref only and so you don't have to calculate it again?